### ISSUE TITLE
Language Card Compact Overflow

### ISSUE THREAD

**Issue by WilsontheWolf** [anuraghazra/github-readme-stats#2619]
### Describe the bug

On the language card, when its compact, long language names (such as "Game Maker Language") overflow out of the card. 

### Expected behaviour

Should not overflow.

### Screenshots / Live demo link

Live:
![](https://github-readme-stats.vercel.app/api/top-langs/?username=WilsontheWolf&&show_icons=true&theme=radical&count_private=true&layout=compact)

Screenshot: 
![image](https://user-images.githubusercontent.com/33164598/232353294-e916b634-d33e-4fc6-bfec-90b00b4ae289.png)

### Additional context

In the past, I remember  this being wider. Maybe that would be a solution. Otherwise some kind of text cutoff would probably be best.

**Comment by LokeshXs** [comment]
can I work on this ^^^^

**Comment by akshayaureatelabs** [comment]
@WilsontheWolf @LokeshXs Hello,

To fix the issue of long language names overflowing out of the language card when it is in compact mode, you can try one of the following solutions:

- Increase the width of the card: You mentioned that in the past, the language card was wider, so increasing the width of the card could help to avoid the overflow issue.

- Use a text cutoff: You can also try using a text cutoff to truncate the long language names and show an ellipsis (...) at the end of the text to indicate that it has been cut off. This will ensure that the text stays within the card and does not overflow. You can add a tooltip or hover effect to show the full language name when a user hovers over the truncated text.

- Use a smaller font size: Another solution is to reduce the font size of the language name when it is too long to fit within the card. However, this may not be an ideal solution as it can make the text hard to read for some users.

**Comment by LokeshXs** [comment]
@akshayaureatelabs, can this be fixed by some CSS like the word-wrap property?
 
 @WilsontheWolf I can increase the default size of the card but if the user wants a wider card the width option can be exclusively passed. 
Like -[https://github-readme-stats.vercel.app/api/top-langs/?username=WilsontheWolf&layout=compact&card_width=400](url)

**Comment by akshayaureatelabs** [comment]
> @akshayaureatelabs, can this be fixed by some CSS like the word-wrap property?
> 
> @WilsontheWolf I can increase the default size of the card but if the user wants a wider card the width option can be exclusively passed. Like -[https://github-readme-stats.vercel.app/api/top-langs/?username=WilsontheWolf&layout=compact&card_width=400](url)

@LokeshXs Yes you can!

**Comment by rickstaa** [comment]
For now, the best thing is to use the `card_width` option. However, feel free to open a pull request for me to review. 

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=wilsonthewolf&layout=compact)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=wilsonthewolf&layout=compact)](https://github.com/anuraghazra/github-readme-stats)
```


[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=wilsonthewolf&layout=compact&card_width=400)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=wilsonthewolf&layout=compact&card_width=400)](https://github.com/anuraghazra/github-readme-stats)
```

**Comment by Sasikumar00** [comment]
Or we can set the width of the "rect" relative to the "main-card-body". We can get the current width of the main-card-body and then set the "rect' width to be 5-10% more than that

**Comment by ForsakenRei** [comment]
I noticed the same thing when there is some longer names on the right side, currently I just manually set the `card_width`.
If there is some dynamic adjustment to the card it will be great since I don't know if some day the ranking changed and I will need adjust it again or not.

**Comment by rickstaa** [comment]
> I noticed the same thing when there is some longer names on the right side, currently I just manually set the `card_width`. If there is some dynamic adjustment to the card it will be great since I don't know if some day the ranking changed and I will need adjust it again or not.
> 
> [#2732 (comment)](https://github.com/anuraghazra/github-readme-stats/issues/2732#issuecomment-1596835371) This is exactly the way to go.

Hey @ForsakenRei. To my knowledge, this issue has yet to be addressed. I think the second solution I opted in that comment would be the preferred one since it requires less code:

> When card_width is specified, use this width while adhering to the current minimum card_width. The too-long language can then be truncated (i.e. Jupyter or ProgramingLang...).

@LokeshXs are you still working on this issue, or can I remove the assignment 🤔?

**Comment by ForsakenRei** [comment]
yes I know it's not implemented yet, though I thought the two bullet points combined is the solution, like if else conditions based on whether `card_width` is set or not and is it smaller than the default minimum card_width or not? I'm not a JS expert so I might understand the implementation wrong.

**Comment by rickstaa** [comment]
> Hi @rickstaa, Yes I am working on this issue, looks like I achieved the solution
> 
> Screenshot: ![image](https://user-images.githubusercontent.com/77314004/283470797-9549d447-d816-4116-a596-adfb2d3a0a32.png)
> 
> Live: [![image](https://user-images.githubusercontent.com/77314004/283470797-9549d447-d816-4116-a596-adfb2d3a0a32.png)](https://githubstats-ten.vercel.app/api/top-langs/?username=t-montes&layout=compact)
> 
> What I did is calculate the width of SVG in case of a compact layout
> 
> * If the longest language name is overflowing the SVG, I changed the width so that the longest language name will be visible.
> * In case there is no language name overflowing, I keep it the same.

Ah, amazing to hear that you managed to solve the problem! 🚀 My only concern with your current solution is that it overwrites the `card_width` value when a user sets it. Maybe use the ellipsis (i.e. `...`) when the user has specified the `card_width` and otherwise enlarged the card?

**Comment by LokeshXs** [comment]
Yes it overwrites if the content gets out of the box.  If we add an ellipse it would hide the percentage value. since its a important piece of info should we do it?

**Comment by LokeshXs** [comment]
Hi, @rickstaa ,

I have developed a solution for a text truncation problem, but it's not working properly when the user specifies the width of the text. 
Currently, I have set a fixed width for the text, and when it exceeds that width, it gets truncated with an ellipsis.
However, I am unable to make the text width adjustable based on the width of SVG. I would appreciate some guidance and assistance with resolving this issue.

Live:

[http://localhost:3000/top-langs/?username=WilsontheWolf&layout=compact](![image](https://github.com/anuraghazra/github-readme-stats/assets/77314004/206735b2-47be-41d6-b461-94374414017e))
